nevergrad/functions/games/game.py

```diff
@@ -0,0 +1,437 @@
+                return 0
+            elif not cards1:
+                return 2
+            elif not cards2:
+                return 1
+
+            # Pick up first card.
+            card1 = cards1[0]
+            card2 = cards2[0]
+            del cards1[0]
+            del cards2[0]
+            stack += [card1, card2]
+
+            # Standard case.
+            if card1 > card2:
+                cards1 += self.war_decide(policy1, len(cards1), sorted(stack, reverse=True))
+                stack = []
+                continue
+            if card2 > card1:
+                cards2 += self.war_decide(policy2, len(cards2), sorted(stack, reverse=True))
+                stack = []
+                continue
+
+            # Battle!
+            if len(cards1) < 2 or len(cards2) < 2:
+                return 1 if len(cards2) < len(cards1) else 2 if len(cards1) < len(cards2) else 0
+
+            stack += [cards1[0], cards2[0]]
+            del cards1[0]
+            del cards2[0]
+        return 0
+
+    def war_decide(self, policy, num_cards, list_of_cards):
+        cards = sorted(list_of_cards, reverse=True)
+        if policy is None:
+            return cards
+        assert len(cards) % 2 == 0
+        a = min(num_cards, 9)  # at most 9
+        b = len(cards) // 2  # at most 26
+        c = cards[0]  # at most 13
+        # print(a, b, c, a*26*13+b*13+c, len(policy))
+        if self.batawaf:
+            seed = policy[a*18*6+b*6+c]  # type: ignore
+        else:
+            seed = policy[a*26*13+b*13+c]  # type: ignore
+        if seed == 0.:
+            return cards
+        state = np.random.RandomState(hash(seed) % (2**32))
+        state.shuffle(cards)
+        return [c for c in cards]
+
+
+# Real life is more complicated! This is a very simple model.
+# pylint: disable=too-many-instance-attributes,too-many-arguments,too-many-statements,too-many-locals
+class Game(inst.InstrumentedFunction):
+    """
+    Parameters
+    ----------
+    nint intaum_stocks: number of stocks to be managed
+    depth: number of layers in the neural networks
+    width: number of neurons per hidden layer
+    """
+
+    def __init__(self, game: str = "war") -> None:
+        self.game = game
+        self.game_object = _Game()
+        the_dimension = self.game_object.play_game(self.game) * 2  # times 2 because we consider both players separately.
+        instrumentation = Instrumentation(inst.var.Array(the_dimension))
+        super().__init__(self._simulate_game, instrumentation)
+        self.instrumentation.probably_noisy = True
+        self.instrumentation.scrambled = True
+        self._descriptors.update(game=game)
+
+    def _simulate_game(self, x: np.ndarray) -> float:
+        # FIXME: an adaptive opponent, e.g. bandit, would be better.
+        # We play a game as player 1.
+        np_state = np.random.get_state()
+        np.random.seed(self.instrumentation.random_state.randint(12560, dtype=np.uint32))
```

Comment:
please avoid those kind of hack for the random seed, there are always dangerous

Reply:
i add a fixme
that will be for january...